x/twap/types/params.go

```diff
@@ -39,12 +48,30 @@ func (p Params) Validate() error {
 		return err
 	}
 
+	if err := validatePeriod(p.RecordHistoryKeepPeriod); err != nil {
+		return err
+	}
+
 	return nil
 }
 
 // Implements params.ParamSet.
 func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
 	return paramtypes.ParamSetPairs{
 		paramtypes.NewParamSetPair(KeyPruneEpochIdentifier, &p.PruneEpochIdentifier, epochtypes.ValidateEpochIdentifierInterface),
+		paramtypes.NewParamSetPair(KeyRecordHistoryKeepPeriod, &p.RecordHistoryKeepPeriod, validatePeriod),
+	}
+}
+
+func validatePeriod(i interface{}) error {
```

Comment:
What is the reason for making this `interface{}` and not a `time.Diration` directly?

Reply:
because functiom `NewParamSetPair` need `func(value interface{}) error` as parameter. 
```
// NewParamSetPair creates a new ParamSetPair instance.
func NewParamSetPair(key []byte, value interface{}, vfn ValueValidatorFn) ParamSetPair {
	return ParamSetPair{key, value, vfn}
}
```

Reply:
```
	ValueValidatorFn func(value interface{}) error
```

Reply:
I hope I'm not missing anything but how is that related exactly?

We call `validatePeriod` from `Params.Validate`:
https://github.com/osmosis-labs/osmosis/blob/2fbde7a263703971c63c41065542cf27c536b9de/x/twap/types/params.go#L51

Where `p.RecordHistoryKeepPeriod` is of type `time.Duration`. Making this `interface{}` and type checking feels redundant because we can be confident about the type of the input.

Reply:
I see now, thanks for pointing this out